test/compression.cc

```diff
@@ -95,6 +95,9 @@ TEST(Compression, fail_inflate) {
     s.next_in = static_cast<Byte*>(static_cast<void*>(&bad[0]));
     s.avail_in = static_cast<unsigned int>(bad.size() * sizeof(std::string::value_type));
   };
+  auto inflate_src_fail3 = [](z_stream& s) {
```

Comment:
minor nit: maybe also specify the return explicitly `[](z_stream& s) -> void` (I've seen `void` type being mentioned in other lambdas).

Reply:
Yes, I'm adding it right now